challenge/contact.py

```diff
@@ -1,188 +0,0 @@
-
-        # perform actions later
-    def __enter__(self):
-        return self
-    def __exit__(self,t,v,s):
-        self.web.quit()
-
-scheduled = set() # deque? avoid duplicate scheduling
-done = set() # web pages already examined
-emails = set() # email addresses found
-
-def normalized(link):
-    url = urlparse(link)
-    urlparts = list(url)
-    if urlparts[0] == 'http' or urlparts[0] == '':
-        urlparts[0] = 'http'
-    urlparts[0] = str.lower(urlparts[0]) # scheme
-    urlparts[1] = str.lower(urlparts[1]) # netloc
-    if 'mailto' == urlparts[0]:
-        urlparts[2] = str.lower(urlparts[2]) # email address in path
-    urlparts[3] = '' # parms
-    urlparts[4] = '' # query
-    urlparts[5] = '' # fragment
-    return urlunparse(urlparts)
-
-class Links:
-    """ iterator for links on a web page """
-    def __str__(self):
-        return str({'domain': self.domain,'page':self.page,'scheduled':self.found,'emails':self.mail})
-
-    def __init__(self,scan,page,domain):
-        self.scan = scan 
-        self.page = page
-        self.domain = domain
-        self.mail = set()
-        try:
-            #scan.web.manage().timeouts().implicitlyWait(10,TimeUnit.SECONDS)
-            self.scan.web.get(page)
-            # TODO: deal with model dialogs
-            # TODO: check if page really html?
-            # TODO: set focus on page first, before any key commands
-            scan.actions.perform()
-            #pdb.set_trace()
-            aa = '//a[@href]'
-            aas = self.scan.web.find_elements_by_xpath(aa)
-            href = [normalized(a.get_attribute('href')) for a in aas]
-            # links with domain
-            lsd = set([h for h in href if self.domain in h])
-            # relative links (or mailto)
-            rls = set([h for h in href if 'http' not in h])
-            self.mail = set([h[7:] for h in href if 'mailto' in h])
-            self.found = lsd|rls
-        except: # limit catches to HTTP or parsing exceptions?
-            self.found = set() # bad links or selenium exceptions on stale elements ?
-
-    def __iter__(self):
-        return self
-
-    def __next__(self):
-        
-         while (0 < len(self.found)):
-             link = self.found.pop() # str = type(link)
-             url = urlparse(normalized(link))
-             scheme = str(url.scheme)
-             if 0 == len(scheme) or (re.match('http',scheme) and re.search(self.domain+'$',str(url.netloc))):
-                 todo = urlunparse(url)
-                 return todo
-         raise StopIteration
-
-def headers(page):
-    ret = {'Content-Type':None,'ETag':None}
-    try:
-        with urllib.request.urlopen(page) as response:
-             ret = response.info()
-    except:
-        pass # ignore any errors, in particular 404
-    return ret
-
-logging.basicConfig()
-log = logging.getLogger('fe')
-
-def is_html(content_type):
-    if content_type:
-        for mime in ['text/html','application/xhtml+xml']:
```

Comment:
wouldn't this whole method reduce to:
```
return content_type in ['text/html', 'application/xhtml+xml']
```

if content type was None, it wouldn't be in the list.  additionally, do we want to check `mime in content_type` and not that `mime == content_type`?